webapp/channels/jest.config.js

```diff
@@ -41,14 +41,21 @@ const config = {
         'node_modules/(?!react-native|react-router|p-queue|p-timeout|@mattermost/compass-components|@mattermost/compass-icons)',
     ],
     setupFiles: ['jest-canvas-mock'],
-    setupFilesAfterEnv: ['<rootDir>/src/tests/setup.ts'],
+    setupFilesAfterEnv: ['<rootDir>/src/tests/setupTests.ts'],
     testEnvironment: 'jsdom',
     testTimeout: 60000,
-    testURL: 'http://localhost:8065',
+    testEnvironmentOptions: {
+        url: 'http://localhost:8065',
+    },
     watchPlugins: [
         'jest-watch-typeahead/filename',
         'jest-watch-typeahead/testname',
     ],
+    snapshotFormat: {
+        escapeString: true,
+        printBasicPrototype: true,
+    },
+    workerIdleMemoryLimit: 0.8,
```

Comment:
open to suggestion https://jestjs.io/docs/next/configuration#workeridlememorylimit-numberstring

Reply:
That's hilarious that their fix for "Jest runners are leaking memory" is "kill runners that leak enough memory".

Is there a reason for setting the limit to specifically 80% of system memory?

Reply:
This is running on public infrastructure so we do not actually care . We can be greedy there as long as we do not get OOM issues .

Reply:
I'd suggest to gather more data before making a decision. You might want to look at this PR https://github.com/mattermost/mattermost/pull/25031 which adds an action which allows you to look into the system performance during the action. That will print out CPU usage, mem usage etc. 

With that you can dive deep into how much memory is actually being used to set the right thresholds.

Reply:
Perf here
- https://github.com/mattermost/mattermost/pull/25297 without upgrade
- https://github.com/mattermost/mattermost/pull/25296 with upgrade

Reply:
Great idea, Agniva. That one on the branch looks much healthier since it's actually running that whole time. The one on the master branch has so much CPU time taken up by the system with what I assume is swapping memory onto disk. 

I'm good leaving this amount as-is then since it seems to be a clear improvement on CI at least. We can always tweak the memory further later on if we find we need to